stl/inc/xstring

```diff
@@ -2432,6 +2440,106 @@ public:
         _Proxy._Release();
     }
 
+    basic_string(_String_constructor_concat_tag, const basic_string& _Source_of_al, const _Elem* const _Left_ptr,
+        const size_type _Left_size, const _Elem* const _Right_ptr, const size_type _Right_size)
+        : _Mypair(
+            _One_then_variadic_args_t(), _Alty_traits::select_on_container_copy_construction(_Source_of_al._Getal())) {
+        _STL_INTERNAL_CHECK(_Left_size <= max_size()); // caller must check for emission of length_error

```

Comment:
Ambiguous comment - does the caller need to check whether this constructor throws `length_error`, or (what you intended) does the caller need to perform validation and throw `length_error` for bad inputs instead of calling this constructor? I recommend rephrasing to either talking about a precondition, or `sizes must be validated before calling this constructor` or something.

Reply:
I just removed the comment, I think the assert speaks for itself.